### ISSUE TITLE
LS takes >10min to initialize

### ISSUE THREAD

**Issue by Trenrod** [elastic/logstash#6114]
- Version: RC1 5.0.0
- Operating System: Ubuntu LTS 16.04
- Config File (if you have sensitive info, please remove it):
Using everything as default after installing as debian package. (Service not running)
sudo -u logstash /usr/share/logstash/bin/logstash --path.settings=/etc/logstash -e 'input { stdin { } } output { stdout {} }'
- Sample Data: none
- Steps to Reproduce:
1) Install Elasticsearch (rc1) from documentation as debian package
2) Install Kibana (rc1) from documentation as debian package
3) Install Logstash (rc1) from documentation as debian package
4) run: sudo -u logstash /usr/share/logstash/bin/logstash --path.settings=/etc/logstash -e 'input { stdin { } } output { stdout {} }'

 - It takes at least 10minutes till the stdout says that the input is ready.
 - CPU usage is at 0.6 ~ 6.0% (overall usage is at max 10%)
 - Same is happening when starting as a service. First response after "Started logstash." is about 10minutes later. Even on invalid configurations.

**Comment by suyograo** [comment]
when you do `sudo service start logstash`, i.e. starting LS as a service, how long does it take?

**Comment by Trenrod** [comment]
Always around 10 Minutes. Even with invalid configuration

**Oct 25 16:51:22** vps287768 systemd[1]: Started logstash.
**Oct 25 17:01:10** vps287768 logstash[13584]: INFO: Logstash requires a setting file which is typically located in $LS_HOME/config or /etc/logstash. If you installed Logstash through a package and are starting it manually please specify the location to this settings file by passing in "--path.settings=/path/.."
Oct 25 17:01:10 vps287768 logstash[13584]: ERROR: Failed to load settings file from "path.settings". Aborting... path.setting=/etc/logstash, exception=ArgumentError, message=>Setting "path.log" hasn't been registered


Oct 25 16:51:22 vps287768 systemd[1]: Started logstash.
Oct 25 17:01:10 vps287768 logstash[13584]: INFO: .....

**Comment by Trenrod** [comment]
Another hint maybe. If I try to stop the service
- systemctl stop logstash

It always crashes:
Oct 25 17:27:41 vps287768 systemd[1]: Started logstash.
Oct 25 17:30:28 vps287768 systemd[1]: Stopping logstash...
Oct 25 17:30:29 vps287768 systemd[1]: logstash.service: Main process exited, code=exited, status=143/n/a
Oct 25 17:30:29 vps287768 systemd[1]: Stopped logstash.
Oct 25 17:30:29 vps287768 systemd[1]: logstash.service: Unit entered failed state.
Oct 25 17:30:29 vps287768 systemd[1]: logstash.service: Failed with result 'exit-code'.

**Comment by untergeek** [comment]
I see the hostname as `vps287768`.  What kind of host is this? What hardware? What version of Java?

**Comment by jsvd** [comment]
can you also check if https://github.com/jruby/jruby/wiki/Improving-startup-time#ensure-your-system-has-adequate-entropy can be your problem?

**Comment by Trenrod** [comment]
- Its  a virtual private server.
- 4GB RAM, Single CPU@2.4Ghz, SSD (I benchmarked the hardware several times)
- About Java
openjdk version "1.8.0_91"
OpenJDK Runtime Environment (build 1.8.0_91-8u91-b14-3ubuntu1~16.04.1-b14)
OpenJDK 64-Bit Server VM (build 25.91-b14, mixed mode)

**Comment by untergeek** [comment]
I just tested this on a vagrant box (16.04.1, which is running on top of a FreeBSD 11.0 box), with a spinning disk.  Check out the times:

```
vagrant@packer-virtualbox-iso-1474648640:~$ cat /etc/os-release
NAME="Ubuntu"
VERSION="16.04.1 LTS (Xenial Xerus)"
ID=ubuntu
ID_LIKE=debian
PRETTY_NAME="Ubuntu 16.04.1 LTS"
VERSION_ID="16.04"
HOME_URL="http://www.ubuntu.com/"
SUPPORT_URL="http://help.ubuntu.com/"
BUG_REPORT_URL="http://bugs.launchpad.net/ubuntu/"
UBUNTU_CODENAME=xenial
vagrant@packer-virtualbox-iso-1474648640:~$ date
Tue Oct 25 15:52:44 UTC 2016
vagrant@packer-virtualbox-iso-1474648640:~$ /usr/share/logstash/bin/logstash --version
logstash 5.0.0
vagrant@packer-virtualbox-iso-1474648640:~$ sudo -u logstash /usr/share/logstash/bin/logstash --path.settings=/etc/logstash -e 'input { stdin { } } output { stdout {} }'
hello world
Sending Logstash logs to /var/log/logstash which is now configured via log4j2.properties.
The stdin plugin is now waiting for input:
2016-10-25T15:53:17.731Z packer-virtualbox-iso-1474648640 hello world
```

Pre-start time: Tue Oct 25 15:52:44 UTC 2016
Message through time: 2016-10-25T15:53:17.731Z

Considering I'm virtualized doubly here (FreeBSD's Linux compatibility layer + (Vagrant + VirtualBox)), I'd say that a 33 second start time isn't too bad.  I'm in agreement with @jsvd here that there's probably something going south with entropy.  We haven't seen these kinds of startup delays in our testing (almost all of which is done in VMs).

**Comment by Trenrod** [comment]
Solved.
Its now at about ~20 seconds

Fixed by "Setup Additional Entropy for Cloud Servers Using Haveged"
https://www.digitalocean.com/community/tutorials/how-to-setup-additional-entropy-for-cloud-servers-using-haveged

**Comment by jordansissel** [comment]
This issue should be solved out-of-the-box when any of the following are released: Logstash 6.0.0, 5.5.0, and 5.6.0 -- via #7469
